### PR TITLE
added angular2 boilerplate

### PR DESCRIPTION
**Pull request by daryltanwk**

### CODE REVIEW

File: frontend/.angular-cli.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  "apps": [
+    {
+      "root": "src",
+      "outDir": "dist",
+      "assets": [
+        "assets",
+        "favicon.ico"
+      ],
+      "index": "index.html",
+      "main": "main.ts",
```
Comment (soedar) [inline frontend/.angular-cli.json]: ooo typescript, sexy

File: frontend/.angular-cli.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    },
+    {
+      "project": "src/tsconfig.spec.json"
+    },
+    {
+      "project": "e2e/tsconfig.e2e.json"
+    }
+  ],
+  "test": {
+    "karma": {
```
Comment (soedar) [inline frontend/.angular-cli.json]: hmm i guess using 2 different test libraries is okay. is there a way for the boilerplate to use mocha tests instead?

File: frontend/package.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "psycho-web",
+  "version": "0.0.0",
+  "license": "MIT",
```
Comment (soedar) [inline frontend/package.json]: should probably be consistent with the repo license

**Comment by soedar** [comment]
so much boilerplate :|

File: frontend/.angular-cli.json (hunk 2 above)
Comment (daryltanwk) [inline frontend/.angular-cli.json]: i dont know how to do this. but probably possible.

**Pull request merged**